torch/csrc/distributed/autograd/context/context.cpp

```diff
@@ -63,24 +66,32 @@ DistAutogradContext::recvFunctions() const {
 
 void DistAutogradContext::accumulateGrad(
     const torch::autograd::Variable& variable,
-    const torch::Tensor& grad) {
+    const torch::Tensor& grad,
+    size_t num_expected_refs) {
   TORCH_INTERNAL_ASSERT(grad.defined());
   TORCH_INTERNAL_ASSERT(variable.requires_grad());
 
   std::lock_guard<std::mutex> guard(lock_);
   auto it = accumulatedGrads_.find(variable);
+  at::Tensor old_grad;
   if (it != accumulatedGrads_.end()) {
     // Accumulate multiple grads on the same variable.
-    it->value().add_(grad);
-  } else {
-    // First grad for this variable.
-    if (grad.is_sparse()) {
-      accumulatedGrads_.insert(variable, grad.clone());
-    } else {
-      accumulatedGrads_.insert(
-          variable, grad.clone(at::MemoryFormat::Contiguous));
-    }
+    old_grad = it->value();
   }
+
+  // No higher order gradients supported in distributed autograd.
+  AutoGradMode grad_mode(false);
```

Comment:
That doesn't look like the right place to enforce this?
Is this because the AccumulateGrad is not a Node here?

Reply:
I guess if there is no plan in supporting `create_graph=True`, that is ok.